### PR TITLE
Add CMakeLists.txt from feature/ufs of fv3-bundle and extend for SOCA…

### PR DESCRIPTION
**Pull request by aerorahul**
This PR:
- adds a `CMakeListst.txt` to build UFS with JEDI while maintaining options for Atmosphere, DataAtm+Ocean and Coupled UFS apps.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+ExternalProject_Add(ufs-weather-model
+  PREFIX ${CMAKE_CURRENT_BINARY_DIR}/ufs-weather-model
+  GIT_REPOSITORY https://github.com/mark-a-potts/ufs-weather-model.git
+  GIT_SUBMODULES_RECURSE TRUE
+  GIT_TAG develop
+  SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/ufs-weather-model
+  INSTALL_DIR ${DEPEND_LIB_ROOT}
+  CMAKE_ARGS ${UFS_WEATHER_MODEL_ARGS}
+  CMAKE_ARGS -DCCPP_SUITES=FV3_GFS_v16,FV3_GFS_v16_coupled,FV3_GFS_v15p2 -DREPRO=on -DOPENMP=ON -DINLINE_POST=OFF -DMULTI_GASES=OFF -DMPI=ON -DAPP=ATM -DCMAKE_INSTALL_PREFIX=${DEPEND_LIB_ROOT}
```
Comment (rsdunlapiv) [inline CMakeLists.txt]: Maybe `-DAPP=${UFS_APP}`

**Review by rsdunlapiv — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (aerorahul) [reply CMakeLists.txt]: good catch.  I failed to add that in the `CMAKE_ARGS`.

**Review by guillaumevernieres — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (aerorahul) [reply CMakeLists.txt]: Done in 3dbd3df

Comment (aerorahul) [reply CMakeLists.txt]: `CCPP_SUITES` will need to be updated based on the `UFS_APP` as well.  But that can be handled in another PR later.

**Pull request merged**